### PR TITLE
MGMT-8096: Remove all VSphere platform properties

### PR DESCRIPTION
**Pull request by michaellevy101**
# Assisted Pull Request

## Description
VSphere administrators won't share their env data on the cloud, so removing all the vsphere platform specific properties

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by michaellevy101** [comment]
@?

File: deploy/assisted-service-service.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: v1
+x§apiVersion: v1
```
Comment (avishayt) [inline deploy/assisted-service-service.yaml:1]: :scream:

File: swagger.yaml [hunk 2]
```diff
@@ -6201,40 +6201,6 @@ definitions:
     type: object
     x-go-custom-tag: gorm:"embedded;embeddedPrefix:vsphere_"
     description: Vsphere platform specific configuration upon which to perform the installation
-    properties:
```
Comment (avishayt) [inline swagger.yaml:6204]: Why not remove all of vsphere_platform, including the reference from "platform"?

File: internal/provider/vsphere/installConfig.go [hunk 3]
```diff
@@ -5,31 +5,17 @@ import (
...
-		platform.Cluster = PhCluster
-		platform.VCenter = PhVcenter
-		platform.Network = PhNetwork
-		platform.DefaultDatastore = PhDefaultDatastore
-		platform.Username = PhUsername
-		platform.Password = PhPassword
-		platform.Datacenter = PhDatacenter
-	}
+func setPlatformValues(platform *installcfg.VsphereInstallConfigPlatform) {
+	// This is to support adding parameters in day2
```
Comment (avishayt) [inline internal/provider/vsphere/installConfig.go:11]:
```suggestion
	// Add placeholders to make it easier to replace in day2
```

File: internal/provider/vsphere/installConfig.go [hunk 4]
```diff
@@ -5,31 +5,17 @@ import (
...
-		// This is to support adding parameters in day2
-		platform.Cluster = PhCluster
-		platform.VCenter = PhVcenter
-		platform.Network = PhNetwork
-		platform.DefaultDatastore = PhDefaultDatastore
-		platform.Username = PhUsername
-		platform.Password = PhPassword
-		platform.Datacenter = PhDatacenter
-	}
+func setPlatformValues(platform *installcfg.VsphereInstallConfigPlatform) {
```
Comment (romfreiman) [inline internal/provider/vsphere/installConfig.go:10]: hm... why do we need it? We should ask for creds neither.

File: internal/provider/vsphere/installConfig.go (hunk 3 above)
Comment (michaellevy101) [reply internal/provider/vsphere/installConfig.go:11]: Changed, but it's the original code, I just removed the if section

File: deploy/assisted-service-service.yaml (hunk 1 above)
Comment (michaellevy101) [reply deploy/assisted-service-service.yaml:1]: nice catch, thanks

File: internal/provider/vsphere/installConfig.go (hunk 4 above)
Comment (michaellevy101) [reply internal/provider/vsphere/installConfig.go:10]: It's just a placeholders

Comment (romfreiman) [reply internal/provider/vsphere/installConfig.go:10]: sorry, thought that it's for day2

**Pull request merged**